src/components/ebay-carousel/index.js

```diff
@@ -205,12 +280,30 @@ function handleDotClick(originalEvent, target) {
     emitAndFire(this, 'carousel-slide', { slide: slide + 1, originalEvent });
 }
 
+/**
+ * Toggles the play state of an autoplay carousel.
+ *
+ * @param {MouseEvent} originalEvent
+ */
+function togglePlay(originalEvent) {
+    const { state: { config, paused } } = this;
+    config.preserveItems = true;
+    this.setState('paused', !paused);
+    if (paused) this.nextEl.click();
```

Comment:
Did you intend for the `carousel-next` event to be called by this (and the other time .click() is used)? I think we only wanted to call those events from human interaction, so that they can be used for tracking purposes.

Reply:
In the code above it is triggered by clicking the play button and so I think it makes sense, for the actual autoplay part I am not sure, should we have a separate event? Should it not emit any special events? cc @ianmcburnie @senthilp?

Reply:
I would prefer:

```js
if (paused) {
   this.gotoNextSlide(); // (or watever our method is called to move to next slide)
}
```

 i.e. call our public or internal API rather than simulate click.

Reply:
Yeah I agree. Okay I'll make this change.